sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
This method is de facto used 'initializer'. Adding check for header changes seemed better than just reverting PR #277 which seems to be useful.